src/components/structures/MatrixChat.js

```diff
@@ -1271,8 +1281,11 @@ export default createReactClass({
         // particularly noticeable when there are lots of 'limited' /sync responses
         // such as when laptops unsleep.
         // https://github.com/vector-im/riot-web/issues/3307#issuecomment-282895568
-        cli.setCanResetTimelineCallback(function(roomId) {
+        cli.setCanResetTimelineCallback(async function(roomId) {
             console.log("Request to reset timeline in room ", roomId, " viewing:", self.state.currentRoomId);
+            // TODO is there a better place to plug this in
```

Comment:
There's the `Room.timelineReset` event on the client you could listen for. It's probably best to avoid making this callback async as well.

Reply:
I tried that event but I don't think it fired reliably, or maybe it doesn't like multiple event listeners? I can retry moving it there. Any more info why not make it async?

Reply:
> I tried that event but I don't think it fired reliably

Hmm, interesting. It does [seem to be fired unconditionally](https://github.com/matrix-org/matrix-js-sdk/blob/develop/src/models/event-timeline-set.js#L224) in EventTimelineSet. Any particular case you found it didn't fire?

> Any more info why not make it async?

Just feels like this "informational" callback shouldn't have too many side-effects, let alone async ones. There's a slight risk of introducing race conditions as well in case multiple syncs would come in fast. And it would need some plumbing in the js-sdk, as the callback currently isn't expected to return a promise.

Reply:
I don't really remember the specifics anymore, might have imagined it or done something stupid. Will change it up and see if it works.